kratos/utilities/coordinate_transformation_utilities.h

```diff
@@ -465,95 +465,7 @@ class CoordinateTransformationUtils {
 	virtual void Rotate(TLocalVectorType& rLocalVector,
 			GeometryType& rGeometry) const
 	{
-		//const unsigned int LocalSize = rLocalVector.size(); // We expect this to work both with elements (4 nodes) and conditions (3 nodes)
-
-		//unsigned int Index = 0;
-
-		if (rLocalVector.size() > 0)
-		{
-			if(mBlockSize != mDomainSize) //Monolithic case
-			{
-				for(unsigned int j = 0; j < rGeometry.PointsNumber(); ++j)
-				{
-					if( this->IsSlip(rGeometry[j]) )
-					{
-						if(mDomainSize == 3)
-						{
-							array_1d<double,4> aux,aux1;
-							BoundedMatrix<double,4,4> rRot;
-							LocalRotationOperator3D<4>(rRot,rGeometry[j]);
-
-							for(unsigned int k=0; k<4; k++)
-							aux[k] = rLocalVector[j*mBlockSize+k];
-
-							noalias(aux1) = prod(rRot,aux);
-
-							for(unsigned int k=0; k<4; k++)
-							rLocalVector[j*mBlockSize+k] = aux1[k];
-						}
-						else
-						{
-							array_1d<double,3> aux,aux1;
-							BoundedMatrix<double,3,3> rRot;
-							LocalRotationOperator2D<3>(rRot,rGeometry[j]);
-
-							for(unsigned int k=0; k<3; k++)
-							{
-								aux[k] = rLocalVector[j*mBlockSize+k];
-							}
-
-							noalias(aux1) = prod(rRot,aux);
-
-							for(unsigned int k=0; k<3; k++)
-							rLocalVector[j*mBlockSize+k] = aux1[k];
-						}
-					}
-					//Index += mBlockSize;
-				}
-
-			}
-			else //fractional step case
-			{
-				for(unsigned int j = 0; j < rGeometry.PointsNumber(); ++j)
-				{
-					if( this->IsSlip(rGeometry[j]) )
-					{
-						if(mDomainSize == 3)
-						{
-							array_1d<double,3> aux,aux1;
-							BoundedMatrix<double,3,3> rRot;
-							LocalRotationOperatorPure(rRot,rGeometry[j]);
-
-							for(unsigned int k=0; k<3; k++)
-							aux[k] = rLocalVector[j*mBlockSize+k];
-
-							noalias(aux1) = prod(rRot,aux);
-
-							for(unsigned int k=0; k<3; k++)
-							rLocalVector[j*mBlockSize+k] = aux1[k];
-						}
-						else
-						{
-							array_1d<double,2> aux,aux1;
-							BoundedMatrix<double,2,2> rRot;
-							LocalRotationOperatorPure(rRot,rGeometry[j]);
-
-							for(unsigned int k=0; k<2; k++)
-								aux[k] = rLocalVector[j*mBlockSize+k];
-
-							noalias(aux1) = prod(rRot,aux);
-
-							for(unsigned int k=0; k<2; k++)
-							rLocalVector[j*mBlockSize+k] = aux1[k];
-						}
-					}
-					//Index += mBlockSize;
-				}
-
-			}
-
-		}
-
+        RotateRHSAux(rLocalVector, rGeometry);
```

Comment:
```suggestion
            RotateRHSAux(rLocalVector, rGeometry);
```
Please follow the indentation rules.